app/(main)/league/all/page.test.tsx

```diff
@@ -1,124 +1,235 @@
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+
     render(<Leagues />);
 
-    await waitFor(() => {
-      expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
-    });
+    await waitForElementToBeRemoved(() => screen.getByTestId('global-spinner'));
+
+    expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
   });
-  xtest('should not display GlobalSpinner after loading data', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+
+  test('should handle form submission to join a league', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: { id: '123', leagues: [] },
+      allLeagues: [
+        {
+          leagueId: '123',
+          leagueName: 'Test League',
+          participants: [],
+          survivors: [],
+        },
+      ],
+    });
+
+    mockGetAllLeagues.mockResolvedValueOnce([
+      {
+        leagueId: '123',
+        leagueName: 'Test League',
+        participants: [],
+        survivors: [],
+      },
+    ]);
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+    mockAddUserToLeague.mockResolvedValueOnce({});
 
     render(<Leagues />);
 
-    expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
-
     await waitFor(() => {
       expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
     });
+
+    const selectElement = screen.getByTestId('select-available-leagues');
+    fireEvent.change(selectElement, { target: { value: '123' } });
+    fireEvent.click(screen.getByText(/Join League/i));
+
+    await waitFor(() => {
+      expect(mockAddUserToLeague).toHaveBeenCalledWith({
+        leagueId: '123',
+        userId: '123',
+      });
+
+      expect(toast.custom).toHaveBeenCalledWith(
+        <Alert
+          variant={AlertVariants.Success}
+          message={`Added Test League to your leagues!`}
+        />,
+      );
+    });
   });
 
-  xtest('should handle form submission to join a league', async () => {
-    mockUseDataStore.mockReturnValueOnce({
+  test('should show error if adding to league fails', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
       user: { id: '123', leagues: [] },
       allLeagues: [
         {
           leagueId: '123',
           leagueName: 'Test League',
-          logo: 'https://findmylogo.com/logo.png',
-          participants: ['123456', '78'],
-          survivors: ['123456', '78', '9'],
+          participants: [],
+          survivors: [],
         },
       ],
     });
+
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+    mockGetAllLeagues.mockResolvedValueOnce([
+      {
+        leagueId: '123',
+        leagueName: 'Test League',
+        participants: [],
+        survivors: [],
```

Comment:
logo is missing from here. can even be an empty string. just want to make sure we are showing the correct data structure.